strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java

```diff
@@ -679,6 +685,38 @@ public void testChecksumsStorage(@MavenRepository(repositoryId = "checksums-stor
         assertThat(actualChecksums).isEqualTo(expectedChecksums);
     }
 
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    public void testFailArtifactStorageWhenChecksumIsBroken(@MavenRepository(repositoryId = "broken-checksum-test")
```

Comment:
@newarcher36 the test case seems not really clear to me, can you please explain what you assumed here?

Reply:
Hi @sbespalov ,  I am assuming that the checksum of strongbox-test-artifact-1.0.jar.md5 differs from the checksum of the recent downloaded artifact. 

For instance, checksum of strongbox-test-artifact-1.0.jar.md5 is "broken_checksum" and the checksum of the downloaded artifact is "10d759d4a8a297bf5e28a6fc7eaf4b969d78a098".
I that case when is a Maven 2 layout, log and throw and ArtifactStorageException.

Reply:
it seems currently it just checks that there was no checksums cached before checksum file stored, and there is also no point of [this](https://github.com/strongbox/strongbox/pull/1577/files#diff-a0ca838f38a43fbc4b2b2c83bf4b1fb5R707) write.

Reply:
the main test case should be like follows:
1. you should not use `repositoryId` for `MavenTestArtifact`, because artifact should be deployed within test case.
2. you should store the artifact first to chache the checksum in `ChecksumCacheManager`
3. you should store the invalid checksum file and receive the exception

there is also should be a test case like you currenty have but without `Files.write`

and final test case should check valid case.

Reply:
Hi @sbespalov, thank you very much for providing your help I already fixed the test following your instructions but I still need to use Files.write in order to write and invalid checksum value in the md5 file and then store this invalid checksum md5 file in order to trigger ArtifactStorageException otherwise I do not know how can I create a file with an invalid checksum...